src/store.ts

```diff
@@ -1,139 +1,162 @@
+import { ModelManager } from "@accordproject/concerto-core";
+import { TemplateMarkInterpreter } from "@accordproject/template-engine";
+import { TemplateMarkTransformer } from "@accordproject/markdown-template";
+import { transform } from "@accordproject/markdown-transform";
 
-import {SAMPLES, Sample} from './samples';
-import * as playground from './samples/playground';
+import { SAMPLES, Sample } from "./samples";
+import * as playground from "./samples/playground";
 
 interface AppState {
-    templateMarkdown: string
-    modelCto: string
-    data: string
-    agreementHtml: string
-    error: string|undefined
-    samples: Array<Sample>
-    sampleName: string;
-    setTemplateMarkdown: (template: string) => Promise<void>
-    setModelCto: (model: string) => Promise<void>
-    setData: (data: string) => Promise<void>
-    rebuild: () => Promise<void>
-    init: () => Promise<void>
-    loadSample: (name:string) => Promise<void>
+  templateMarkdown: string;
+  modelCto: string;
+  data: string;
+  agreementHtml: string;
+  error: string | undefined;
+  samples: Array<Sample>;
+  sampleName: string;
+  setTemplateMarkdown: (template: string) => Promise<void>;
+  setModelCto: (model: string) => Promise<void>;
+  setData: (data: string) => Promise<void>;
+  rebuild: () => Promise<void>;
+  init: () => Promise<void>;
+  loadSample: (name: string) => Promise<void>;
 }
 
 async function rebuild(template: string, model: string, dataString: string) {
-    const modelManager = new ModelManager({ strict: true });
-    modelManager.addCTOModel(model, undefined, true);
-    await modelManager.updateExternalModels();
-    const engine = new TemplateMarkInterpreter(modelManager, {});
+  const modelManager = new ModelManager({ strict: true });
+  modelManager.addCTOModel(model, undefined, true);
+  await modelManager.updateExternalModels();
+  const engine = new TemplateMarkInterpreter(modelManager, {});
 
-    const templateMarkTransformer = new TemplateMarkTransformer();
+  const templateMarkTransformer = new TemplateMarkTransformer();
 
-    const templateMarkDom = templateMarkTransformer.fromMarkdownTemplate({ content: template }, modelManager, 'contract', { verbose: false });
-    // console.log(JSON.stringify(templateMarkDomEn, null, 2));
+  const templateMarkDom = templateMarkTransformer.fromMarkdownTemplate(
+    { content: template },
+    modelManager,
+    "contract",
+    { verbose: false }
+  );
 
-    const data = JSON.parse(dataString);
+  const data = JSON.parse(dataString);
 
-    const ciceroMark = await engine.generate(templateMarkDom, data);
-    // console.log(JSON.stringify(ciceroMark.toJSON(), null, 2));
-    return await transform(ciceroMark.toJSON(), 'ciceromark_parsed', ['html'], {}, { verbose: false });
+  // Check if 'last' property is provided in the data
+  if ("last" in data) {
+    throw new Error(
+      "Error: 'last' is an optional property and should not be provided in the data."
+    );
+  }
```

Comment:
Could you please explain the need for this change?

Reply:
Ohh, Just realized! Was actually fixing an issue on "Type checker doesn't detect usage of optional properties" from part of the issues. Would scrap that out. It is void in this PR.

Reply:
@DianaLease It's definitely good to go now, you could lookup the updated description in the PR for more details. Reviews and comments are welcome.